src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
Per #662, I favor integrating it into the main `text` strategy as a new keyword but @DRMacIver isn't a fan - David, can you just make a BDFL ruling and save us the debate?